rahulsmehta/PairDE
# Initial compilation service implementation

This pull request implements basic features of compilation service:
* Receiving base64-encoded java source file as a POST request
* Decoding the file and saving it to a tmp directory
* Compiling the file, and responding with any compilation errors

Next steps for this are wiring this to the UI to enable compilation results to be displayed in the "console" pane below the code editor. Implementing this is blocked on refactoring the front end to use react-mosaic instead of react-split-panel.

To test:
* To see an example of the "success" case where there are no compilation errors, issue a POST request to /compile with the body `{
	"file_name":"HelloWorld.java",
	"encoded_src":"cHVibGljIGNsYXNzIEhlbGxvV29ybGQgewogICAgcHVibGljIHN0YXRpYyB2b2lkIG1haW4oU3RyaW5nW10gYXJndil7CiAgICAgICAgU3lzdGVtLm91dC5wcmludGxuKCJIZWxsbywgd29ybGQhIik7CiAgICB9Cn0K"
}`
* For the failure case, use this body: `{
	"file_name":"HelloWorld.java",
	"encoded_src":"cHVibGljIGNsYXNzIEhlbGxvV29ybGQgewogICAgcHVibGljIHN0YXRpYyB2b2lkIG1haW4oU3RyaW5nW10gYXJndil7CiAgICAgICAgU3lzdGVtLm91dC5wcmludGxuKCJIZWxsbywgd29ybGQhIikKICAgIH0KfQo="
}`
